stan/math/rev/functor/algebra_solver_powell.hpp

```diff
@@ -222,42 +212,95 @@ Eigen::VectorXd algebra_solver_powell(
  * @throw <code>std::domain_error</code> if the norm of the solution exceeds
  * the function tolerance.
  */
+template <typename F, typename T1, typename... T_Args,
+          require_all_eigen_vector_t<T1>* = nullptr,
+          require_any_st_var<T_Args...>* = nullptr>
+Eigen::Matrix<var, Eigen::Dynamic, 1> algebra_solver_powell_impl(
+    const F& f, const T1& x, std::ostream* msgs, double relative_tolerance,
+    double function_tolerance, long int max_num_steps,
+    const T_Args&... args) {  // NOLINT(runtime/int)
+  const auto& x_eval = x.eval();
+
+  auto arena_args_tuple = std::make_tuple(to_arena(args)...);
+
+  const auto& x_val = (value_of(x_eval)).eval();
+
+  auto args_vals_tuple = std::make_tuple(eval(value_of(args))...);
+
+  // algebra_solver_check(x_val, y, dat, dat_int, function_tolerance,
+  //                     max_num_steps);
+  check_nonnegative("alegbra_solver", "relative_tolerance", relative_tolerance);
+
+  // Construct the Powell solver
+
+<<<<<<< HEAD
```

Comment:
@jgaeb Looks like I messed up the merge but some variant of the variadic stuff we had is uploaded lol